cjetreconstruction/src/cjetreconstruction-finder.cc

```diff
@@ -0,0 +1,284 @@
+  auto njets_option = opts.add<Value<int>>("", "njets", "njets value for exclusive jets");
+  auto dump_option = opts.add<Value<string>>("d", "dump", "Filename to dump jets to");
+  auto debug_clusterseq_option = opts.add<Switch>("c", "debug-clusterseq", "Dump cluster sequence jet and history content");
+  auto gc_option = opts.add<Switch>("", "gc", "Enable garbage collector");
+
+
+
+  opts.parse(argc, argv);
+
+  if (help_option->count() == 1) {
+    cout << argv[0] << " [options] HEPMC3_INPUT_FILE" << endl;
+    cout << endl;
+	  cout << opts << "\n";
+    cout << "Note the only one of ptmin, dijmax or njets can be specified!\n" << endl;
+    exit(EXIT_SUCCESS);
+  }
+
+  const auto extra_args = opts.non_option_args();
+  std::string input_file{};
+  if (extra_args.size() == 1) {
+    input_file = extra_args[0];
+  } else if (extra_args.size() == 0) {
+    std::cerr << "No <HepMC3_input_file> argument after options" << std::endl;
+  } else {
+    std::cerr << "Only one <HepMC3_input_file> supported" << std::endl;
+  }
+
+  // Check we only have 1 option for final jet selection
+  auto sum = int(njets_option->is_set()) + int(dijmax_option->is_set()) + int(ptmin_option->is_set());
+  if (sum != 1) {
+    cerr << "One, and only one, of ptmin, dijmax or njets needs to be specified (currently " <<
+      sum << ")" << endl;
+    exit(EXIT_FAILURE);
+  }
+
+  // read in input events
+  //----------------------------------------------------------
+  auto events = read_input_events(input_file.c_str(), maxevents);
+
+  // Set strategy
+  auto strategy =
+      jetreconstruction_RecoStrategy::JETRECONSTRUCTION_RECOSTRATEGY_BEST;
+  if (mystrategy == string("N2Plain")) {
+    strategy =
+        jetreconstruction_RecoStrategy::JETRECONSTRUCTION_RECOSTRATEGY_N2PLAIN;
+  } else if (mystrategy == string("N2Tiled")) {
+    strategy =
+        jetreconstruction_RecoStrategy::JETRECONSTRUCTION_RECOSTRATEGY_N2TILTED;
+  }
+
+  auto algorithm =
+      jetreconstruction_JetAlgorithm::JETRECONSTRUCTION_JETALGORITHM_ANTIKT;
+  if (alg != "") {
+    if (alg == "AntiKt") {
+      algorithm =
+          jetreconstruction_JetAlgorithm::JETRECONSTRUCTION_JETALGORITHM_ANTIKT;
+      power = -1.0;
+    } else if (alg == "CA") {
+      algorithm =
+          jetreconstruction_JetAlgorithm::JETRECONSTRUCTION_JETALGORITHM_CA;
+      power = 0.0;
+    } else if (alg == "Kt") {
+      algorithm =
+          jetreconstruction_JetAlgorithm::JETRECONSTRUCTION_JETALGORITHM_KT;
+      power = 1.0;
+      //} else if (alg == "GenKt") {
+      //  algorithm = fastjet::genkt_algorithm;
+      //} else if (alg == "Durham") {
+      //  algorithm = fastjet::ee_kt_algorithm;
+      //  power = 1.0;
+      //} else if (alg == "EEKt") {
+      //  algorithm = fastjet::ee_genkt_algorithm;
+    } else {
+      std::cout << "Unknown algorithm type: " << alg << std::endl;
+      exit(1);
+    }
+  } else {
+    if (power == -1.0) {
```

Comment:
I would remove the code that drives things by power value only - it won't work when we do support e+e- algorithms